src/stan/analyze/mcmc/ess.hpp

```diff
@@ -0,0 +1,105 @@
+#ifndef STAN_ANALYZE_MCMC_ESS_HPP
+#define STAN_ANALYZE_MCMC_ESS_HPP
+
+#include <stan/math/prim.hpp>
+#include <stan/analyze/mcmc/autocovariance.hpp>
+#include <algorithm>
+#include <cmath>
+#include <vector>
+#include <limits>
+
+namespace stan {
+namespace analyze {
+
+/**
+ * Computes the effective sample size (ESS) for the specified
+ * parameter across all chains.  The number of draws per chain must be > 3,
```

Comment:
I didn't realize we now had this _and_ `compute_effective_sample_size` -- could you comment (here and in the code, preferably) on what the differences are?
Should we mark `compute_effective_sample_size` as deprecated?

Ditto for rhat and `compute_potential_scale_reduction`

Reply:
It seems the CI failures are currently because of these warnings up in cmdstan -- probably worth not doing now, but instead circling back after cmdstan PR is merged. I think it's fine to leave in the docstrings